base/reflection.jl

```diff
@@ -1310,6 +1310,28 @@ function return_types(@nospecialize(f), @nospecialize(types=default_tt(f));
     return rt
 end
 
+function infer_effects(@nospecialize(f), @nospecialize(types=default_tt(f));
+                       world = get_world_counter(),
+                       interp = Core.Compiler.NativeInterpreter(world))
+    ccall(:jl_is_in_pure_context, Bool, ()) && error("code reflection cannot be used from generated functions")
+    types = to_tuple_type(types)
+    effects = Core.Compiler.EFFECTS_TOTAL
```

Comment:
What happens if there's no method match? Does this mean it returns total, even though the function throws?

Reply:
Good catch. We need to account for `MethodError` as we do here:
https://github.com/JuliaLang/julia/blob/5146e352abc48eb6a4ee05789a76031c034e46fa/base/compiler/abstractinterpretation.jl#L219